README.md

```diff
@@ -1,6 +1,463 @@
-# Bring your own Copilot to Terminal Pro
+# Bring your own Copilot to Terminal
 
-Welcome to the example repository for integrating a custom copilot into OpenBB
-Terminal Pro. This repository provides reference implementations to help you
-create and add your own custom copilots to enhance the capabilities of Terminal
-Pro.
+Welcome to the example repository for integrating custom copilots into the OpenBB Terminal.
+
+This repository provides everything you need to build and add your own custom
+copilots to OpenBB Copilot.
+
+Here are a few common reasons why you might want to build your own copilot:
+- You have a unique data source that you don't want to add as a custom integration to OpenBB Terminal.
+- You want to use a specific LLM.
+- You want to use a local LLM.
+- You want a Copilot that is self-hosted on your infrastructure.
+- You are running on-premise in a locked-down environment that doesn't allow data to leave your VPC.
+
+
+## Overview
+
+To integrate a custom copilot, you'll need to create a custom API backend that
+OpenBB Terminal connects to. OpenBB Terminal will make requests to your backend, allowing you to interact with your custom copilot from Terminal Pro.
+
+Your custom API backend will respond with Server-Sent Events
+([SSEs](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events)).
+
+**Note: If you're looking to get started
+quickly, we suggest running one of the example copilots included as part of
+this repository, and adding it as a custom copilot to OpenBB Terminal (each example copilot includes instructions on how to run them). Cloning and modifying an example copilot is a great way to build a custom copilot.**
+
+## The copilot protocol is stateless
+
+The most important concept to understand is that the copilot protocol is
+_stateless_.  This means that every request from OpenBB Terminal to your copilot
+will include all previous messages (such as AI completions, human messages,
+function calls, and function call results) in the request payload.
+
+This means it is not necessary for your custom copilot to maintain any state
+between requests. It can simply use the request payload to generate a response.
+
+OpenBB Terminal is responsible for maintaining the conversation state, and will
+append the responses to the `messages` array in the request payload. You can
+choose how much of the conversation you want your custom copilot to generate a
+response to, but typically you'd respond to the entire conversation history.
+
+We recommend that you do not maintain any conversational state in your copilot,
+and simply respond to the entire conversation history.
```

Comment:
I found this section confusing.

The responses will update the initial message chain received in the payload with responses, it sounds that we are editing the payload of the request.

On choosing "how much of the conversation you want your custom copilot to generate a
response to", makes it sound like there's a setting to choose how much of the message chain will be considered, but it's very specific to the implementation. Think it's too specific for the README.